openmsupply/open-msupply
# 2103 link contact to patient

Part of #2103

# 👩🏻‍💻 What does this PR do? 

Add a "Link contact to mSupply patient" modal to the contact tracing page.

![image](https://github.com/openmsupply/open-msupply/assets/88299195/58415262-9314-4283-a78b-030c0c859b26)

![image](https://github.com/openmsupply/open-msupply/assets/88299195/2475757a-3a7d-4c7c-82df-984201e1fda0)

This is the very base implementations and has a couple of shortcomings/ could be improved:
- For some reason search input fields loose focus after search which is super annoying, any ideas why?
- Not able to unlink patient
- Not able to create new patient (you first have to create the patient and the link it)
- UI design: e.g. button into top right?
- Better visualization needed if patient is linked or not
- Use same UI or reuse components from the patient creation modal
- Fetch patient from central when not found locally (see point above)

# Testing 

- Create a contact trace entry
- Optional: Enter some contact details under "Family Partner information"
- Click the new "Link to patient" button
- If contact details have been entered the search fields should be prepopulated
- Search for a patient and select it
- The icon of the "Link to patient" button should be a check now, indicating that the patient is linked (better visualization needed)
- Relink to different patient
- Save and reload
